control_plane/src/storage.rs

```diff
@@ -341,11 +342,29 @@ impl PageServerNode {
 
     pub fn tenant_create(
         &self,
-        new_tenant_id: Option<ZTenantId>,
+        tenant_id: Option<ZTenantId>,
+        settings: HashMap<&str, &str>,
```

Comment:
Up to you, but I'd rather had some method to parse the strings into `TenantConfOpt` (maybe put the toml parsing into that module too and add some unit tests for both?) and pass it around, not some odd map.